proposals/0303-swiftpm-extensible-build-tools.md

```diff
@@ -921,13 +921,14 @@ There is inherent risk in running build tools provided by other packages. This c
 This proposal is intentionally fairly basic and leaves many improvements for the future. Among them are:
 
 - the ability for package plugins to define new types that can be used to configure those plugins in package manifests
-- the ability for a build tool to have depend on different packages than the clients of the plugin that uses it
+- the ability for a build tool to have dependencies on different versions of packages than the clients of the plugin that uses it
 - the ability for plugins to have access to the full build graph at a detailed level
 - the ability for prebuild actions to depend on tools built by SwiftPM
 - the ability for package plugin scripts to use libraries provided by other targets
 - the ability for build commands to emit output files that feed back into the rule system to generate new work during the build (this requires support in SwiftPM's native build system as well as in the build systems of some IDEs that use libSwiftPM)
 - the ability to provide per-target type-safe options to a use of an plugin (details below)
 - the ability to define post-build command and other kinds of commands
+- specific support for testing plugins (they can currently be testing using test fixtures that exercise the plugins, but it would be useful to be able to invoke them directly, for example through a `swift package` command that invokes the plugin with specific inputs)
```

Comment:
yeah good future direction 👍 

I'll have some info and hints to drop here once we get to this :-) In sbt this is done by https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html so maybe some inspiration to collect from it once we get to it